### PR TITLE
Add cypress test

### PR DESCRIPTION
**Pull request by LePetitTim**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+image: docker:stable
```
Comment (submarcos) [inline .gitlab-ci.yml:1]: you use docker image here, but you used docker-compose below

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+stages:
+  - e2e-tests
+
+cypress:
+  stage: e2e-tests
+  tags: [visu-cicd-docker]
+  image: docker/compose:latest
```
Comment (submarcos) [inline .gitlab-ci.yml:8]: i think docker-compose is better

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  services:
+    - docker:dind
+
+  before_script:
+    - chown -R 1000:1000 .
+    - apk update
+    - apk upgrade
+    - apk add python3 python3-dev build-base gcc openssl-dev libc-dev make git docker-compose
```
Comment (submarcos) [inline .gitlab-ci.yml:16]: no need to install python3 python3-dev and docker-compose in a docker/compose image, it's already available (maybe other libs installed here are already available)

File: cypress.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (submarcos) [inline cypress.json:1]: this file is not required because it's already defined in cypress folder

File: cypress/cypress.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "fixturesFolder": "fixtures",
+  "supportFile": "support",
+  "integrationFolder": "integration",
+  "pluginsFile": false,
+  "screenshotsFolder": "screenshots",
+  "baseUrl": "http://front:3000",
```
Comment (submarcos) [inline cypress/cypress.json:7]: use http://localhost:3000 as baseUrl, then prefix test launchig with BASEURL=http://front:3000 to let developpers run test on their computers

**Comment by submarcos** [comment]
@jrmi @LePetitTim As geocrud frontend is just an instance of terra-admin, I suggest to build a terra-backend default app to run e2e test directly in terra-admin CI, then add these tests to validate CRUD part of

File: cypress/compose.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+version: "3"
+services:
+  cypress:
+    environment:
+      - CYPRESS_baseUrl=http://front
+      # - DISPLAY
+    image: cypress/included:4.9.0
```
Comment (submarcos) [inline cypress/compose.yml:7]: I suggest to use included:latest to always keep and test latest version of cypress and browsers included to test ou app

(and maybe run test with differents browsers ? (firefox / chrome and chromium-edge ?)

Comment (submarcos) [reply cypress/compose.yml:7]: https://github.com/Terralego/terra-admin/issues/262

File: cypress/compose.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+version: "3"
+services:
+  cypress:
+    environment:
+      - CYPRESS_baseUrl=http://front
```
Comment (jrmi) [inline cypress/compose.yml:5]: Prefer CYPRESS_BASE_URL as this env var version seems deprecated.

Comment (submarcos) [reply cypress/compose.yml:5]: +1

**Comment by LePetitTim** [comment]
ReDo it with githubactions

**Review by submarcos — APPROVED** [review]

**Pull request merged**